logpyle/__init__.py

```diff
@@ -1219,6 +1220,38 @@ def __call__(self):
         return result
 
 
+class InitTime(LogQuantity):
```

Comment:
It's a inelegant IMO that this produce a whole pile of data (one entry per time step, per rank) to record one scalar.

Reply:
I agree, but IMHO the other options are worse:

- Using this as a quantity (instead of e.g. a constant) allows easier rank reduction, plotting, etc.
- With the change to `return None`, this only creates one value per rank in the sqlite database.

Reply:
> Using this as a quantity (instead of e.g. a constant) allows easier rank reduction, plotting, adding to watches, etc.

- It's not useful as a watch.
- Rank reduction and plotting (just not vs time) can easily be done with constants.

> one value per rank in the sqlite database (which is what we want I think).

And a few billion `NULL` values. Which aren't free to encode.

Reply:
> * Rank reduction and plotting (just not vs time) can easily be done with constants.

Sure, but the interface is different, right?

> > one value per rank in the sqlite database (which is what we want I think).
> 
> And a few billion `NULL` values. Which aren't free to encode.

Are you sure? From what I can see, logpyle does not store values for `None` in the db. See also: https://github.com/illinois-ceesd/logpyle/blob/43c6060bdfea3140e0ab91e546dd141d1511fa3e/logpyle/__init__.py#L650-L665

Reply:
> Sure, but the interface is different (compared to "normal" quantities), right?

No, not really. You just don't use a leading `$`.

> Are you sure? From what I can see, logpyle does not store values for None in the db.

You're right. I forgot that it uses one table per quantity.

Reply:
(I still think it should be a constant though. :)

Reply:
> > Sure, but the interface is different (compared to "normal" quantities), right?
> 
> No, not really. You just don't use a leading `$`.

How would I do a reduction/rank aggregation?

```python
>>> .q select $t_vis.max       # quantity
t_vis
---------------------
0.05441403388977051
[...]

>>> .q select unixtime.max    # constant
Traceback (most recent call last):
  File "/Users/mdiener/Work/emirge/logpyle/logpyle/runalyzer.py", line 305, in push
    self.execute_magic(cmdline)
  File "/Users/mdiener/Work/emirge/logpyle/logpyle/runalyzer.py", line 357, in execute_magic
    self.db.print_cursor(self.db.q(args))
  File "/Users/mdiener/Work/emirge/logpyle/logpyle/runalyzer.py", line 36, in q
    return self.db.execute(self.mangle_sql(qry), extra_args)
sqlite3.OperationalError: no such column: unixtime.max
```

From what I can see, constants don't support rank aggregation, a default rank aggregator, a unit, or description.

Reply:
`select max(t_init) from runs group by run_id`

Reply:
> select max(t_init) from runs group by run_id

As far as I can see, `runs` only has one entry per run (not per rank), how can this do rank aggregation?

Reply:
You're right. Per-rank constants are not currently a thing. I think we should fix that at some point (#47), but I agree that it's not urgent. I'm OK with gathering this as a quantity for now.

Reply:
Also, sorry about being confused twice. :)